src/Tile/Tile.model.js

```diff
@@ -48,7 +48,7 @@ export class Tile extends Model {
 
   static async upsertEstate(estate) {
     return asyncBatch({
-      elements: estate.data.parcels,
+      elements: estate ? estate.data.parcels : [], // Some Estates appear as undefined
```

Comment:
Do we know when this happens? because it looks like it should be a check in the caller. For example adding a 

`if (!asset) return`

on `upsertAsset`

Reply:
I would add it to `upsertAsset`, but also remove the `throw` related to the `assetType`

```javascript
default:
          throw new Error(`The asset type ${assetType} is invalid`)
```

It is weird to not throw if the asset is `undefined` or `null`, but the type is not `parcel` or `estate`:
```javascript
 static async upsertAsset(assetId, assetType) {
    const asset = await Asset.getNew(assetType).findById(assetId)
    if (asset) {
      // Some Estates appear as undefined
      switch (assetType) {
        case ASSET_TYPES.parcel:
          return this.upsertParcel(asset)
        case ASSET_TYPES.estate:
          return this.upsertEstate(asset)
        default:
          throw new Error(`The asset type ${assetType} is invalid`)
      }
    }
  }
```

I suggest, to omit if the asset is not valid: asset itself and type


```javascript
 static async upsertAsset(assetId, assetType) {
    const asset = await Asset.getNew(assetType).findById(assetId)
    if (asset) {
      // Some Estates appear as undefined
      switch (assetType) {
        case ASSET_TYPES.parcel:
          return this.upsertParcel(asset)
        case ASSET_TYPES.estate:
          return this.upsertEstate(asset)
        default:
         break
      }
    }
  }
```

Reply:
I'll try to reproduce on Ropsten